### PR TITLE
grt: Converting TreeNode and TreeEdge pointers to vectors

### PR DESCRIPTION
**Pull request by luis201420**

### CODE REVIEW

File: src/grt/src/fastroute/include/DataType.h [hunk 1]
```diff
@@ -169,7 +169,8 @@ struct TreeNode
   short heights[max_connections];
   int eID[max_connections];
 
-  short x, y;   // position in the grid graph
+  short x, y;  // position in the grid graph
+  int nbrCount = 0;
```
Comment (eder-matheus) [inline src/grt/src/fastroute/include/DataType.h:173]: Use snake_case for member names: `nbr_count`

File: src/grt/src/fastroute/src/maze3D.cpp [hunk 2]
```diff
@@ -895,8 +895,8 @@ void FastRouteCore::mazeRouteMSMDOrder3D(int expand,
 
     int enlarge = expand;
     const int num_terminals = sttrees_[netID].num_terminals;
-    const auto& treeedges = sttrees_[netID].edges;
-    const auto& treenodes = sttrees_[netID].nodes;
+    auto& treeedges = sttrees_[netID].edges;
+    auto& treenodes = sttrees_[netID].nodes;
```
Comment (eder-matheus) [inline src/grt/src/fastroute/src/maze3D.cpp:899]: Why they're losing const? Here and below.

**Review by eder-matheus — CHANGES_REQUESTED** [review]

File: src/grt/src/fastroute/src/maze3D.cpp (hunk 2 above)
Comment (luis201420) [reply src/grt/src/fastroute/src/maze3D.cpp:899]: Because those variables are then modified in the function

Comment (eder-matheus) [reply src/grt/src/fastroute/src/maze3D.cpp:899]: I didn't notice some function calls that use these variables were marked as non-const.

**Review by eder-matheus — APPROVED** [review]
Waiting for secure-ci before merging.

**Pull request merged**